Source/Core/UICommon/ResourcePack/ResourcePack.cpp

```diff
@@ -28,34 +30,46 @@ constexpr char TEXTURE_PATH[] = HIRES_TEXTURES_DIR DIR_SEP;
 
 ResourcePack::ResourcePack(const std::string& path) : m_path(path)
 {
-  auto file = unzOpen(path.c_str());
-  Common::ScopeGuard file_guard{[&] { unzClose(file); }};
+  void* zip_reader = mz_zip_reader_create();
+  if (!zip_reader)
+  {
+    m_valid = false;
+    m_error = "Failed to create zip reader";
+    return;
+  }
+
+  Common::ScopeGuard file_guard{[&] { mz_zip_reader_delete(&zip_reader); }};
 
-  if (file == nullptr)
+  if (mz_zip_reader_open_file(zip_reader, path.c_str()) != MZ_OK)
   {
     m_valid = false;
     m_error = "Failed to open resource pack";
     return;
   }
 
-  if (unzLocateFile(file, "manifest.json", 0) == UNZ_END_OF_LIST_OF_FILE)
+  if (mz_zip_reader_locate_entry(zip_reader, "manifest.json", 0) != MZ_OK)
   {
     m_valid = false;
     m_error = "Resource pack is missing a manifest.";
     return;
   }
 
-  unz_file_info64 manifest_info{};
-  unzGetCurrentFileInfo64(file, &manifest_info, nullptr, 0, nullptr, 0, nullptr, 0);
+  mz_zip_file* manifest_info;
+  if (mz_zip_reader_entry_get_info(zip_reader, &manifest_info) != MZ_OK)
+  {
+    m_valid = false;
+    m_error = "Failed to access manifest.json";
+    return;
+  }
 
-  std::string manifest_contents(manifest_info.uncompressed_size, '\0');
-  if (!Common::ReadFileFromZip(file, &manifest_contents))
+  std::string manifest_contents(manifest_info->uncompressed_size, '\0');
+  if (!Common::ReadFileFromZip(zip_reader, &manifest_contents))
   {
     m_valid = false;
     m_error = "Failed to read manifest.json";
     return;
   }
-  unzCloseCurrentFile(file);
+  mz_zip_reader_entry_close(zip_reader);
```

Comment:
This `mz_zip_reader_entry_close` call seems out of place to me. It doesn't seem to pair with an `mz_zip_reader_entry_open` call?